src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -194,16 +196,31 @@ public override void WriteObject(object graph, Type type, ODataMessageWriter mes
                 // the next page link is not set when calling WriteStart(feed) but is instead set later on that feed
                 // object before calling WriteEnd(), the next page link will be written at the end, as required for
                 // odata.streaming=true support.
-                if (nextPageLink != null)
+                if (nextLinkGenerator != null)
```

Comment:
>nextLinkGenerator [](start = 20, length = 17)

nextLinkGenerator would always be non-null, right?  I think you intend to check nextLinkGenerator(lastObject) != null